src/bench/crypto_hash.cpp

```diff
@@ -1,248 +1,189 @@
-// Copyright (c) 2016 The Bitcoin Core developers
+// Copyright (c) 2016-2020 The Bitcoin Core developers
 // Copyright (c) 2018-2020 The Dash Core developers
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <iostream>
 
 #include <bench/bench.h>
-#include <bloom.h>
-#include <hash.h>
-#include <random.h>
-#include <uint256.h>
-#include <utiltime.h>
 #include <crypto/ripemd160.h>
 #include <crypto/sha1.h>
 #include <crypto/sha256.h>
 #include <crypto/sha3.h>
 #include <crypto/sha512.h>
+// #include <crypto/siphash.h>
+#include <hash.h>
+#include <random.h>
+#include <uint256.h>
 
 /* Number of bytes to hash per iteration */
 static const uint64_t BUFFER_SIZE = 1000*1000;
 
-static void HASH_RIPEMD160(benchmark::State& state)
+static void RIPEMD160(benchmark::Bench& bench)
 {
     uint8_t hash[CRIPEMD160::OUTPUT_SIZE];
     std::vector<uint8_t> in(BUFFER_SIZE,0);
-    while (state.KeepRunning())
+    bench.batch(in.size()).unit("byte").run([&] {
         CRIPEMD160().Write(in.data(), in.size()).Finalize(hash);
+    });
 }
 
-static void HASH_SHA1(benchmark::State& state)
+static void SHA1(benchmark::Bench& bench)
 {
     uint8_t hash[CSHA1::OUTPUT_SIZE];
     std::vector<uint8_t> in(BUFFER_SIZE,0);
-    while (state.KeepRunning())
+    bench.batch(in.size()).unit("byte").run([&] {
         CSHA1().Write(in.data(), in.size()).Finalize(hash);
+    });
 }
 
-static void HASH_SHA256(benchmark::State& state)
+static void SHA256(benchmark::Bench& bench)
 {
     uint8_t hash[CSHA256::OUTPUT_SIZE];
     std::vector<uint8_t> in(BUFFER_SIZE,0);
-    while (state.KeepRunning())
+    bench.batch(in.size()).unit("byte").run([&] {
         CSHA256().Write(in.data(), in.size()).Finalize(hash);
+    });
 }
 
-static void HASH_SHA3_256_1M(benchmark::State& state)
+static void SHA3_256_1M(benchmark::Bench& bench)
 {
     uint8_t hash[SHA3_256::OUTPUT_SIZE];
     std::vector<uint8_t> in(BUFFER_SIZE,0);
-    while (state.KeepRunning())
+    bench.batch(in.size()).unit("byte").run([&] {
         SHA3_256().Write(in).Finalize(hash);
+    });
 }
 
-static void HASH_SHA256_0032b(benchmark::State& state)
+static void SHA256_32b(benchmark::Bench& bench)
 {
     std::vector<uint8_t> in(32,0);
-    while (state.KeepRunning()) {
+    bench.batch(in.size()).unit("byte").run([&] {
         CSHA256()
             .Write(in.data(), in.size())
             .Finalize(in.data());
-    }
-}
-
-static void HASH_DSHA256(benchmark::State& state)
```

Comment:
The various-byte tests were not present in Bitcoin so they were dropped, the only exception made was for X11.

Reply:
>The various-byte tests were not present in Bitcoin

Well, that's because they were implemented to compare performance to similar X11 benchmarks. Pls bring them back and revert the removal of the `HASH_` prefix.